Userland/Shell/Tests/builtin-test.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+source $(dirname "$0")/test-commons.inc
+
+
+if not [ "$(type ls)" = "ls is $(which ls)" ] { fail "'type' on a binary not working" }
+
+if not [ "$(type pwd)" = "pwd is a shell builtin" ] { fail "'type' on a builtin not working" }
+
+f() { ls }
+
+if not [ "$(type f)" = "f is a function\nf() {\nls\n}" ] { fail "'type' on a function not working" }
```

Comment:
Now this is failing, I think it must be a bug in `[` or something else :thinking:.